File: src/java/kotlin-test/src/androidTest/java/com/tencent/wcdbtest/orm/testclass/AutoAddColumnObject.kt
// Created by qiuwenchen on 2023/6/25.
//

/*
 * Tencent is pleased to support the open source community by making
 * WCDB available.
 *
 * Copyright (C) 2017 THL A29 Limited, a Tencent company.
 * All rights reserved.
 *
 * Licensed under the BSD 3-Clause License (the "License"); you may not use
 * this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 *       https://opensource.org/licenses/BSD-3-Clause
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.tencent.wcdbtest.orm.testclass

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBTableCoding

@WCDBTableCoding
class AutoAddColumnObject {
    @WCDBField(isPrimary = true)
    var primaryValue = 0

    @WCDBField(isUnique = true)
    var uniqueValue = 0

    @WCDBField
    var insertValue = 0

    @WCDBField
    var updateValue = 0

    @WCDBField
    var selectValue = 0

    @WCDBField
    var multiSelectValue = 0

    @WCDBField
    var deleteValue = 0

    @WCDBField
    var indexValue = 0
}

File: src/rust/wcdb_rust/tests/orm/testclass/auto_add_column_object.rs
use table_coding::WCDBTableCoding;

#[derive(WCDBTableCoding)]
pub struct AutoAddColumnObject {
    #[WCDBField(column_name = "primaryValue", is_primary = true)]
    primary_value: i32,
    #[WCDBField(column_name = "uniqueValue", is_unique = true)]
    unique_value: i32,
    #[WCDBField(column_name = "insertValue")]
    insert_value: i32,
    #[WCDBField(column_name = "updateValue")]
    update_value: i32,
    #[WCDBField(column_name = "selectValue")]
    select_value: i32,
    #[WCDBField(column_name = "multiSelectValue")]
    multi_select_value: i32,
    #[WCDBField(column_name = "deleteValue")]
    delete_value: i32,
    #[WCDBField(column_name = "indexValue")]
    index_value: i32,
}

impl AutoAddColumnObject {
    pub fn new() -> Self {
        Self {
            primary_value: 0,
            unique_value: 0,
            insert_value: 0,
            update_value: 0,
            select_value: 0,
            multi_select_value: 0,
            delete_value: 0,
            index_value: 0,
        }
    }
}
